Formula/vineyard.rb

```diff
@@ -0,0 +1,110 @@
+  depends_on "tbb"
+
+  fails_with gcc: "5"
+
+  resource "libclang" do
+    url "https://files.pythonhosted.org/packages/ea/ec/94fefe778caa8f2ecf9bb996917535a49b36580846af908b2f38fe6396c9/libclang-14.0.1.tar.gz"
+    sha256 "332e539201b46cd4676bee992bbf4b3e50450fc17f71ff33d4afc9da09cf46cb"
+  end
+
+  def install
+    # We install the libclang from pypi for build as the `clang` package
+    # bundled in LLVM cannot find libclang.dylib easily.
+    #
+    # The requirement for libclang python package will be removed once
+    # LLVM been upgrade to 15, see also: Homebrew/homebrew-core#106925
+    venv = virtualenv_create(buildpath/"venv", "python3")
+    venv.pip_install resource("libclang")
+
+    # make libclang.so/libclang.dll findable by clang.cindex.Index.create()
+    ENV["LIBCLANG_LIBRARY_PATH"] = Formula["llvm"].opt_lib
+
+    # link against system libc++ instead of llvm provided libc++
+    ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
+
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DCMAKE_CXX_STANDARD=14",
+                    "-DCMAKE_CXX_STANDARD_REQUIRED=TRUE",
+                    "-DPYTHON_EXECUTABLE=#{buildpath}/venv/bin/python",
+                    "-DUSE_EXTERNAL_ETCD_LIBS=ON",
+                    "-DUSE_EXTERNAL_TBB_LIBS=ON",
+                    "-DUSE_EXTERNAL_NLOHMANN_JSON_LIBS=ON",
+                    "-DBUILD_VINEYARD_TESTS=OFF",
+                    "-DOPENSSL_ROOT_DIR=#{Formula["openssl@1.1"].opt_prefix}",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"test.cc").write <<~EOS
+      #include <iostream>
+      #include <memory>
+
+      #include <vineyard/client/client.h>
+
+      int main(int argc, char **argv) {
+        vineyard::Client client;
+        VINEYARD_CHECK_OK(client.Connect(argv[1]));
+
+        std::shared_ptr<vineyard::InstanceStatus> status;
+        VINEYARD_CHECK_OK(client.InstanceStatus(status));
+        std::cout << "vineyard instance is: " << status->instance_id << std::endl;
+
+        return 0;
+      }
+    EOS
+
+    system ENV.cxx, "test.cc", "-std=c++17",
+                    "-I#{Formula["apache-arrow"].include}",
+                    "-I#{Formula["boost"].include}",
+                    "-I#{include}",
+                    "-I#{include}/vineyard",
+                    "-I#{include}/vineyard/contrib",
+                    "-L#{Formula["apache-arrow"].lib}",
+                    "-L#{Formula["boost"].lib}",
+                    "-L#{lib}",
+                    "-larrow",
+                    "-lboost_thread-mt",
+                    "-lboost_system-mt",
+                    "-lvineyard_client",
+                    "-o", "test_vineyard_client"
+
+    # prepare vineyardd
+    vineyardd_pid = fork do
+      exec bin/"vineyardd",
+        "--norpc",
+        "--meta=local",
+        "--socket=#{testpath}/vineyard.sock"
+    end
```

Comment:
```suggestion
    vineyardd_pid = fork do
      exec bin/"vineyardd", "--norpc",
                            "--meta=local",
                            "--socket=#{testpath}/vineyard.sock"
    end
```

or, this might work also, and is simpler:

```suggestion
    vineyardd_pid = spawn bin/"vineyardd", "--norpc",
                                           "--meta=local",
                                           "--socket=#{testpath}/vineyard.sock"
```